pkg/resources/connection.go

```diff
@@ -0,0 +1,313 @@
+		for _, enableToAccount := range enableFailoverConfig {
+			accountInConfig := enableToAccount.(string)
+			accountIdentifier := sdk.NewAccountIdentifierFromFullyQualifiedName(accountInConfig)
+
+			enableFailoverToAccountsList = append(enableFailoverToAccountsList, accountIdentifier)
+		}
+
+		err := client.Connections.Alter(ctx, sdk.NewAlterConnectionRequest(id).
+			WithEnableConnectionFailover(*sdk.NewEnableConnectionFailoverRequest().
+				WithToAccounts(enableFailoverToAccountsList)),
+		)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return ReadContextConnection(false)(ctx, d, meta)
+}
+
+func ReadContextConnection(withExternalChangesMarking bool) schema.ReadContextFunc {
+	return func(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+		client := meta.(*provider.Context).Client
+		id, err := sdk.ParseAccountObjectIdentifier(d.Id())
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		connection, err := client.Connections.ShowByID(ctx, id)
+		if err != nil {
+			if errors.Is(err, sdk.ErrObjectNotFound) {
+				d.SetId("")
+				return diag.Diagnostics{
+					diag.Diagnostic{
+						Severity: diag.Warning,
+						Summary:  "Failed to retrieve connection. Target object not found. Marking the resource as removed.",
+						Detail:   fmt.Sprintf("Connection name: %s, Err: %s", id.FullyQualifiedName(), err),
+					},
+				}
+			}
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  "Failed to retrieve connection.",
+					Detail:   fmt.Sprintf("Connection name: %s, Err: %s", id.FullyQualifiedName(), err),
+				},
+			}
+		}
+
+		if withExternalChangesMarking {
+			if err := handleExternalChangesToObjectInShow(d,
+				outputMapping{"is_primary", "is_primary", connection.IsPrimary, connection.IsPrimary, nil},
+			); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+
+		errs := errors.Join(
+			setStateToValuesFromConfig(d, connectionSchema, []string{"is_primary"}),
+			d.Set(FullyQualifiedNameAttributeName, id.FullyQualifiedName()),
+			d.Set(ShowOutputAttributeName, []map[string]any{schemas.ConnectionToSchema(connection)}),
+			d.Set("comment", connection.Comment),
+		)
+		if errs != nil {
+			return diag.FromErr(errs)
+		}
+
+		sessionDetails, err := client.ContextFunctions.CurrentSessionDetails(ctx)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		currentAccountIdentifier := sdk.NewAccountIdentifier(sessionDetails.OrganizationName, sessionDetails.AccountName)
+
+		enableFailoverToAccounts := make([]string, 0)
+		for _, allowedAccount := range connection.FailoverAllowedToAccounts {
+			allowedAccountIdentifier := sdk.NewAccountIdentifierFromFullyQualifiedName(allowedAccount)
+			if currentAccountIdentifier.FullyQualifiedName() == allowedAccountIdentifier.FullyQualifiedName() {
+				continue
+			}
+			enableFailoverToAccounts = append(enableFailoverToAccounts, allowedAccountIdentifier.Name())
```

Comment:
We're not saving the current account to state, which is ok, but what if it's present in the config?

Reply:
talking only about the `failover_to_accounts` list, the current account (which has created that connection) is always present in the list at snowflake. We don't want to always specify that the connection itself is in the list. Although if it is provided in the config, it will fail with snowflake error `The connection cannot be failed over to an account in the same region` We could prevent that from happening by adding the same if to Create method or we could enforce that this list should be required and always filled with at least the account of the creator of connection